File: ecommerce-backend/src/core/orderItem.cpp
#include "orderItem.h"

OrderItem::OrderItem(const std::string orderItemId, const Product& product, int quantity)
 : product(product), quantity(quantity), unitPrice(product.getPrice){
    orderItemId = UUIDGenerator::generateUUID();
 }

 const Product& OrderItem::getProduct() const {
     return product;
 }

    int OrderItem::getQuantity() const {
        return quantity;
    }
    double OrderItem::getUnitPrice() const {
        return unitPrice;
    }
    
    double OrderItem::getTotalPrice() const {
        return product.getPrice() * quantity;
    }

File: JAVA/orderitem.java
public class OrderItem {
    private String orderItemId;
    private String productId;
    private int quantity;
    private double price;

    public OrderItem(String productId, int quantity, double price) {
        this.orderItemId = UUID.randomUUID().toString();
        this.productId = productId;
        this.quantity = quantity;
        this.price = price;
    }

    public double getTotalPrice() {
        return quantity * price;
    }

    // Getter methods
    public String getOrderItemId() {
        return orderItemId;
    }

    public String getProductId() {
        return productId;
    }

    public int getQuantity() {
        return quantity;
    }

    public double getPrice() {
        return price;
    }
}